github/actions_artifacts_test.go

```diff
@@ -0,0 +1,243 @@
+	if err != nil {
+		t.Errorf("Actions.ListWorkflowRunArtifacts returned error: %v", err)
+	}
+
+	want := &ArtifactList{TotalCount: Int64(1), Artifacts: []*Artifact{{ID: Int64(1)}}}
+	if !reflect.DeepEqual(artifacts, want) {
+		t.Errorf("Actions.ListWorkflowRunArtifacts returned %+v, want %+v", artifacts, want)
+	}
+}
+
+func TestActionsService_ListWorkflowRunArtifacts_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.ListWorkflowRunArtifacts(context.Background(), "%", "r", 1, nil)
+	testURLParseError(t, err)
+}
+
+func TestActionsService_ListWorkflowRunArtifacts_invalidRepo(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.ListWorkflowRunArtifacts(context.Background(), "o", "%", 1, nil)
+	testURLParseError(t, err)
+}
+
+func TestActionsService_ListWorkflowRunArtifacts_notFound(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/runs/1/artifacts", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	artifacts, resp, err := client.Actions.ListWorkflowRunArtifacts(context.Background(), "o", "r", 1, nil)
+	if err == nil {
+		t.Errorf("Expected HTTP 404 response")
+	}
+	if got, want := resp.Response.StatusCode, http.StatusNotFound; got != want {
+		t.Errorf("Actions.ListWorkflowRunArtifacts return status %d, want %d", got, want)
+	}
+	if artifacts != nil {
+		t.Errorf("Actions.ListWorkflowRunArtifacts return %+v, want nil", artifacts)
+	}
+}
+
+func TestActionsService_GetArtifact(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		fmt.Fprint(w, `{"id":1, "node_id": "xyz", "name": "a", "size_in_bytes": 5, "archive_download_url": "u", "expired": false, "created_at": "b", "expires_at": "c"}`)
+	})
+
+	artifact, _, err := client.Actions.GetArtifact(context.Background(), "o", "r", 1)
+	if err != nil {
+		t.Errorf("Actions.GetArtifact returned error: %v", err)
+	}
+
+	want := &Artifact{ID: Int64(1), NodeID: String("xyz"), Name: String("a"), SizeInBytes: Int64(5), ArchiveDownloadURL: String("u"), Expired: Bool(false), CreatedAt: String("b"), ExpiresAt: String("c")}
+	if !reflect.DeepEqual(artifact, want) {
+		t.Errorf("Actions.GetArtifact returned %+v, want %+v", artifact, want)
+	}
+}
+
+func TestActionsService_GetArtifact_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.GetArtifact(context.Background(), "%", "r", 1)
+	testURLParseError(t, err)
+}
+
+func TestActionsService_GetArtifact_invalidRepo(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.GetArtifact(context.Background(), "%", "r", 1)
```

Comment:
What is this actually testing?  I think it can be deleted.

Reply:
I saw other test files have cases to sanity check parameters (for [example](https://github.com/google/go-github/blob/85084c0ce71a3d798a13cf693714cb9b7719a98e/github/teams_test.go#L43)).  This is just to make sure whatever is given for either the `owner` or `repo` can be URL-parsed.  Looks like I forgot to switch the parameter for some of the test functions, thus why they look like duplicates.  Given that, do you still prefer to have these tests removed?

Reply:
> Given that, do you still prefer to have these tests removed?

I'm totally fine to keep the tests if they are not duplicates and they are testing something. :smile:
Thanks!

Reply:
Sure thing.

Reply:
Fixed.

Reply:
OK, I'm not actually sure what changed here. Are all the tests actually testing something?

Reply:
Yes.  These are sanity checks for the `owner` and `repo` parameters in each function.  They should not longer be duplicates as they were all updated in my latest commit.

Reply:
Here is a link to each test case in the file view:
[GetArtifact_invalidOwner](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L113)
[GetArtifact_invalidRepo](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L121)
[DownloadArtifact_invalidOwner](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L173)
[DownloadArtifact_invalidRepo](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L181)
[DeleteArtifact_invalidOwner](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L245)
[DeleteArtifact_invalidRepo](https://github.com/atorr/go-github/blob/3aa2f08320e9e36105cd19a11b5e02f674212a3b/github/actions_artifacts_test.go#L253)